### PR TITLE
Ensure that depth is not mismatched

### PR DESCRIPTION
**Pull request by Shane32**

### CODE REVIEW

File: src/GraphQLParser/ParserContext.Parse.cs [hunk 1]
```diff
@@ -25,6 +26,9 @@ public GraphQLDocument ParseDocument()
             );
             _document.Definitions = definitions;
             _document.UnattachedComments = _unattachedComments;
+
+            Debug.Assert(_currentDepth == 1, "Depth has not returned to 1 after parsing document");
```
Comment (Shane32) [inline src/GraphQLParser/ParserContext.Parse.cs:30]: This is not compiled within release builds, but will reliable fail tests if the number of `IncreaseDepth()` and `DecreaseDepth()` calls are mismatched.

Comment (sungam3r) [reply src/GraphQLParser/ParserContext.Parse.cs:30]: 👍

**Comment by Shane32** [comment]
@sungam3r There already is a bug in the code.  Trying to track it down.

File: src/GraphQLParser/ParserContext.Parse.cs [hunk 2]
```diff
@@ -1327,7 +1331,6 @@ private GraphQLType ParseType()
                 return type;
             }
 
-            IncreaseDepth();
```
Comment (sungam3r) [inline src/GraphQLParser/ParserContext.Parse.cs:1330]: In fact, this line should not be removed but I add needed calls in other places later.

**Review by sungam3r — APPROVED** [review]

**Pull request merged**